src/librustdoc/html/render.rs

```diff
@@ -246,6 +247,48 @@ struct IndexItem {
     path: String,
     desc: String,
     parent: Option<ast::DefId>,
+    search_type: Option<IndexItemFunctionType>,
+}
+
+/// A type used for the search index.
+struct Type {
+    name: Option<String>,
+}
+
+impl fmt::Display for Type {
+    /// Formats type as {name: $name}.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{{"));
+        try!(write!(f, "\"name\":\"{}\"", self.name.clone().unwrap_or("UNKNOWN".to_string())));
```

Comment:
Perhaps! I was thinking though that we may want to use `null` here instead of `UNKNOWN` (as searching for UNKNOWN probably isn't what you want).

I wonder, could entries with `None` be omitted entirely from the search index?